go-mode.el

```diff
@@ -632,6 +632,94 @@ case keyword. It returns nil for the case line itself."
        ;; line or the previous character is a comma or colon.
        (or (bolp) (looking-back "[,:]" (1- (point)))))))))
 
+(defun go--fill-prefix ()
+  "Return fill prefix for following comment paragraph."
+  (save-excursion
+    (beginning-of-line)
+
+    ;; Skip over empty lines and empty comment openers/closers.
+    (while (and
+            (or (go--empty-line-p) (go--boring-comment-p))
+            (zerop (forward-line 1))))
+
+    ;; If we are looking at the start of an interesting comment, our
+    ;; prefix is the comment opener and any space following.
+    (if (looking-at (concat go--comment-start-regexp "[[:space:]]*"))
+        ;; Replace "/*" opener with spaces so following lines don't
+        ;; get "/*" prefix.
+        (replace-regexp-in-string "/\\*" "  "
+                                  (match-string-no-properties 0)))))
+
+(defun go--fill-paragraph (&rest args)
+  "Wrap fill-paragraph to set custom fill-prefix."
+  (let ((fill-prefix (go--fill-prefix))
+        (fill-paragraph-function nil))
+    (apply 'fill-paragraph args)))
+
+(defun go--empty-line-p ()
+  (looking-at "[[:space:]]*$"))
+
+(defun go--boring-comment-p ()
```

Comment:
Can you add a comment explaining what a boring comment is?

Also its a little confusing that there is both a boring-comment-p and an interesting-comment-p function. One might reasonably expect those to be mutually exclusive when executed from within a comment.